yb-voyager/cmd/analyzeSchema.go

```diff
@@ -1240,22 +1249,24 @@ func packAndSendAnalyzeSchemaPayload(status string, errorMsg string) {
 		if !lo.ContainsBy(reasonsToSendObjectNameToCallhome, func(r string) bool {
 			return strings.Contains(issue.Reason, r)
 		}) {
-			issue.ObjectName = "XXX" // Redacting object name before sending in case reason is not in list
+			issue.ObjectName = constants.OBFUSCATE_STRING // Redacting object name before sending in case reason is not in list
 		}
 		for _, sensitiveReason := range reasonsIncludingSensitiveInformationToCallhome {
-			if strings.Contains(issue.Reason, sensitiveReason) {
-				switch sensitiveReason {
-				case UNSUPPORTED_DATATYPE, UNSUPPORTED_DATATYPE_LIVE_MIGRATION:
-					//e.g. Reason "Unsupported datatype - xml on column - data"
-					//sending only "Unsupported datatype - xml"
-					issue.Reason = strings.Split(issue.Reason, "on column -")[0]
-				default:
-					issue.Reason = sensitiveReason
+			if strings.HasPrefix(issue.Reason, sensitiveReason) {
+				issue.Reason = sensitiveReason
+			} else {
+				match, err := utils.MatchesFormatString(sensitiveReason, issue.Reason)
+				if match {
+					issue.Reason, err = utils.ObfuscateFormatDetails(sensitiveReason, issue.Reason, constants.OBFUSCATE_STRING)
```

Comment:
This obsfuscation business is getting complex. As we were discussing on call, it might not be very useful to even send the reason in future. 

Let's remember to revisit this later.

Reply:
yeah, all that might be affecting the callhome/ybd result output/UI.

Let me add a TODO comment here if thats fine? @makalaaneesh @priyanshi-yb